src/main/js/controls/Timeline/TimelineConfig.js

```diff
@@ -56,6 +96,10 @@ export const processInput = (input, config) => {
     config.bindLegendTo = input.bindLegendTo;
     config.padding = getPadding(config, input.padding);
     config.padding.hasCustomPadding = utils.isDefined(input.padding);
+    config.legendPadding = getLegendPadding(config, input.legendPadding);
+    config.padding.hasCustomLegendPadding = utils.isDefined(
+        input.legendPadding
+    );
```

Comment:
Why do we require this? Is this just boolean value whether `hasCustomLegendPadding` as true or false

Can we update something like this? If we just store the value whether we want to know whether custom legend padding is applied or not
```js
config.padding.hasCustomLegendPadding = config.legendPadding.hasCustomLegendPadding
```

Reply:
It just sets hasCustomLegendPadding property in legendPadding Object to either true or false based on whether legendPadding is provided. Its supposed to be config.legendPadding.hasCustomLegendPadding  instead of config.padding.hasCustomLegendPadding. I have updated it [here](https://github.com/JananiGunasekaran/carbon-graphs/commit/473d2980f065f426b1e398edd34ff91593fb4f86). I have retained this expression just to be consistent with the code structure becasue, for padding, there is [this expression](https://github.com/cerner/carbon-graphs/blob/master/src/main/js/controls/Timeline/TimelineConfig.js#L58) which sets the bool hasCustomPadding.

Reply:
I mean `getLegendPadding(config, input.legendPadding)` method will return the object which will have the following output and We are adding the output to `config.legendPadding`. It already has `hasCustomLegendPadding` prop.

```js
       {
            top: 5px,
            bottom: 5px,
            right: 5px,
            left: 5px,
            hasCustomLegendPadding: false
      }
```

Why do we need this again?
```js
    config.padding.hasCustomLegendPadding = utils.isDefined(
        input.legendPadding
    );
```

Reply:
Removed it [here](https://github.com/JananiGunasekaran/carbon-graphs/commit/2a8d93f94e635278c5f37231389f99bdb3e5c78b).